proposals/0383-allow-let-property-wrapper.md

```diff
@@ -0,0 +1,179 @@
+var height: Int {
+  get { return _height.wrappedValue }
+}
+
+private let _weight: SmallNumber = SmallNumber(wrappedValue: 1)
+var weight: Int {
+  get { return _weight.wrappedValue }
+}
+```
+and results in code that is easy to write and understand:
+```
+struct UnitRectangle {
+  @SmallNumber let height: Int = 1
+  @SmallNumber let width: Int = 1
+}
+```
+
+Property wrappers with `let` declarations will be allowed both as members and local declarations, as envisioned by SE-0258 for `var` declared property wrappers. All other property wrapper traits also remain unchanged from SE-0258.
+
+## Detailed design
+
+Here are three examples of how a `let` wrapped property can make current iterations more effortless.
+
+The `Clamping` property wrapper from [SE-0258 Property Wrappers](https://github.com/apple/swift-evolution/blob/main/proposals/0258-property-wrappers.md#clamping-a-value-within-bounds) can be rewritten to use `let` as its storage and implementation do not change, no matter its application.
+```
+struct Color {
+  @Clamping(min: 0, max: 255) let purple: Int = 127
+}
+```
+
+A `let` wrapped property could be useful for reference types like a property wrapper class. Typically property wrappers are written for value types but occasionally a protocol like `NSObject` may require the use of a class. For example:
+```
+@propertyWrapper
+class WrapperClass<T> : NSObject {
+  var wrappedValue: T
+
+  init(wrappedValue: T) {
+    self.wrappedValue = wrappedValue
+  }
+}
+
+class C {
+  @WrapperClass let value: Int
+
+  init(v: Int) {
+    value = v
+  }
+}
+```
+Here, `WrapperClass` can be made an immutable property wrapper class instance, preventing any future unintentional changes to the property wrapper class type in this context.
+
+SwiftUI property wrappers may also benefit from a let declaration. For example, `@ScaledMetric` in its simplest usage can be written with a `let` instead:
+```
+struct ContentView: View {
+  @ScaledMetric let imageSize = 10
+
+  var body: some View {
+    Image(systemName: "heart.fill")
+    .resizable()
+    .frame(width: imageSize, height: imageSize)
+  }
+}
+```
+
+Similarly, other SwiftUI property wrappers could be `let` declared when they do not require more than a single initialization.
+
+### Property wrappers with `nonmutating set`
+
+Property wrappers that have a `wrappedValue` property with a `nonmutating set` (e.g., SwiftUI's [`@State`](https://developer.apple.com/documentation/swiftui/state/wrappedvalue) and [`@Binding`](https://developer.apple.com/documentation/swiftui/binding/wrappedvalue)) will preserve the reference semantics of the `wrappedValue` implementation even when marked as a `let` declaration. For example:  
+
+```
+@State let weekday: String = "Monday"
+```
+Here, `weekday` is an immutable instance of the `@State` property wrapper, but its `wrappedValue` storage will retain its mutability and reference type traits. This will translate to:
```

Comment:
Neither `State` nor `String` is a reference type. How could this storage have "reference type traits"? Do you mean to say that the property wrapper will retain its reference _semantics_?